### DISCUSSION TITLE
Tailwind Dropdown menu Auto align not working?

### DISCUSSION THREAD

**Discussion by saiengagebay** [tailwindlabs/tailwindcss#13641] [Help]
How can I replicate the Bootstrap 5 dropdown's automatic alignment in the Tailwind Dropdown, as it's not functioning as anticipated?

**Comment by wongjn** [comment]
What do you mean by "the Tailwind Dropdown"? Please consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the undesired behavior you are experiencing. How would you like this dropdown aligned?

**Reply by saiengagebay** [reply]
https://play.tailwindcss.com/25XcR6hTEq

Updated Tailwind dropdown code not working auto alignment.

**Reply by wongjn** [reply]
Oh I see, you're talking about a Tailwind UI component, not Tailwind. So with this extra information and the screenshot of the Bootstrap example you provided in https://github.com/tailwindlabs/tailwindcss/discussions/13641#discussioncomment-9331100, it seems you'd like for the dropdown menu itself to go up or down depending on its position in the viewport.

If you are using the HTML template, you'd need to adjust your JavaScript to work as you'd like – Tailwind UI does not come with JavaScript out of the box. If you are using the React or Vue versions, you may need to have some extra logic to again, swap the direction around when required.

**Reply by saiengagebay** [reply]
Yes, the Tailwind UI dropdown isn't functioning as expected in both React and Vue versions. Specifically, the dropdown menu should dynamically adjust its direction (up or down) based on its position within the viewport. Please refer to the attached screenshot for clarification. [](https://snipboard.io/LYF1ef.jpg)

**Reply by wongjn** [reply]
Ah OK, then it would seem you would need to implement that behavior in your own project.

**Reply by saiengagebay** [reply]
The implementation needs to be integrated directly into the Tailwind UI framework to ensure universal accessibility and usability for all users.

**Comment by saiengagebay** [comment]
Please check the ref bootstrap screenshot: https://snipboard.io/Shb9U4.jpg
